core/ai.go

```diff
@@ -82,9 +83,51 @@ type AIModelConfig struct {
 	Currency      string  `json:"currency,omitempty"`
 }
 
+// UnmarshalJSON allows `PricePerUnit` to be specified as a string.
+func (s *AIModelConfig) UnmarshalJSON(data []byte) error {
+	type Alias AIModelConfig
+	aux := &struct {
+		PricePerUnit interface{} `json:"price_per_unit"`
+		*Alias
+	}{
+		Alias: (*Alias)(s),
+	}
+
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	// Handle PricePerUnit
+	var price JSONRat
+	switch v := aux.PricePerUnit.(type) {
+	case string:
+		pricePerUnit, currency, err := common.ParsePricePerUnit(v)
+		if err != nil {
+			return fmt.Errorf("error parsing price_per_unit: %v", err)
+		}
+		price = JSONRat{pricePerUnit}
+		if s.Currency == "" {
+			s.Currency = currency
+		}
```

Comment:
Hmm I see what you are doing there. Is it worth having this logic compared to just keeping the currency in a separate field in the JSON? This provides multiple ways of doing the same thing and leads to ambiguous behavior (e.g. what happens if I specify both the `"currency"` and add a currency to the `pricePerUnit`?)

AFAIK the go JSON lib (and the `JSONRat` impl) already supports scientific notation, so nothing we need to do add that support there.

If we do go with the new way of specifying the price, I believe we should implement it for the other price config JSON's as well, like the video gateway for price per O.

Reply:
Thinking further about this, maybe we remove this change on the JSON config, which is redundant, and make it only for the CLI flags, wdyt?